<launch>
  <arg name="ns" default="tracking_ros" />
  <arg name="model_type" default="vit_b" />
  <arg name="device" default="cuda:0" />
  
  <arg name="compressed" default="false"/>
  <arg name="input_image" default="/kinect_head/rgb/image_rect_color"/>
  <arg name="input_mask" default="/docker/detic_segmentor/segmentation_info"/>
  <arg name="processed_image" default="/$(arg ns)/processed_image"/>
  <arg name="segmentation_image" default="/$(arg ns)/segmentation_image" />
  <arg name="segmentation_mask" default="/$(arg ns)/segmentation_mask" />
  

  <group ns="$(arg ns)">
    <!-- process image and repub -->
    <node if="$(arg compressed)" name="raw_image_transport" pkg="image_transport" type="republish" args="compressed">
      <remap from="in" to="$(arg input_image)"/>
      <remap from="out" to="$(arg processed_image)"/>
    </node>
    <node unless="$(arg compressed)" name="raw_image_transport" pkg="image_transport" type="republish" args="raw">
      <remap from="in" to="$(arg input_image)"/>
      <remap from="out" to="$(arg processed_image)"/>
    </node>

    <!-- tracking node -->
    <include file="$(find tracking_ros)/launch/detic_node.launch" >
      <arg name="ns" value="$(arg ns)" />
      <arg name="model_type" value="$(arg model_type)" />
      <arg name="device" value="$(arg device)" />
      <arg name="input_mask" value="$(arg input_mask)" />
      <arg name="processed_image" value="$(arg processed_image)"/>
      <arg name="segmentation_image" value="$(arg segmentation_image)" />
      <arg name="segmentation_mask" value="$(arg segmentation_mask)" />
    </include>


    <!-- visualize segmentation -->
    <node name="image_view" pkg="image_view" type="image_view" respawn="false" output="screen">
      <remap from="image" to="$(arg segmentation_image)" />
    </node>

  </group>
</launch>
